src/Context/ContextDefinition.php

```diff
@@ -21,6 +21,7 @@ class ContextDefinition extends ContextDefinitionCore implements ContextDefiniti
     'multiple' => 'isMultiple',
     'required' => 'isRequired',
     'default_value' => 'defaultValue',
+    'form_element' => 'formElement',
```

Comment:
Not that I can see. You need something like:

  /**
   * Type of form element to be used.
   *
   * @var string
   */
  protected $formElement = 'textfield';

ContextDefinition::toArray() is trying to set ContextDefinition::$formElement and it doesn't exist yet.